lib/modules/showImages.js

```diff
@@ -638,292 +602,220 @@ function domainToModuleName(hostname) {
 	return domainStack.pop();
 }
 
-async function checkElementForImage(elem, selfTextExpando = false) {
-	if (module.options.hideNSFW.value) {
-		elem.NSFW = new Thing(elem).isNSFW();
-	} else {
-		elem.NSFW = false;
-	}
-
-	elem.selfTextExpando = selfTextExpando;
-	elem.linklistExpando = !!elem.parentNode.classList.contains('title');
-
-	const href = elem.href;
-	if ((!elem.classList.contains('imgScanned') && (typeof imagesRevealed[href] === 'undefined' || !module.options.ignoreDuplicates.value || isCurrentSubreddit('dashboard')) && href !== null)) {
-		elem.classList.add('imgScanned');
-		dupeAnchors++;
-
-		let detectResult;
-
-		const siteModule = [
-			'default',
-			'defaultVideo',
-			'defaultAudio',
-			domainToModuleName(elem.hostname),
-		]
-			.map(modId => domainModuleMap[modId])
-			.find(mod => mod && (detectResult = mod.detect(elem)));
-
-		if (!siteModule) return;
-
-		elem.site = siteModule.moduleID;
+async function convertGifToVideo(options) {
+	try {
+		const info = await ajax({
+			type: 'json',
+			url: '//upload.gfycat.com/transcodeRelease',
+			data: { fetchUrl: options.src },
+			cacheFor: DAY,
+		});
 
-		if (detectResult && !elem.NSFW) {
-			imagesRevealed[href] = dupeAnchors;
+		if (!info.gfyName) throw new Error('gfycat transcode did not contain "gfyName"');
 
-			try {
-				const options = await siteModule.handleLink(href, detectResult);
-				await createImageExpando(elem, options);
-			} catch (e) {
-				console.error(`showImages: error detecting image expando for ${elem.href}`);
-				console.error(e);
-			}
-		}
-	} else if (!elem.classList.contains('imgScanned')) {
-		const textFrag = document.createElement('span');
-		textFrag.setAttribute('class', 'RESdupeimg');
-		removeRedditExpandoButton(elem);
-		$(textFrag).html(`<a class="noKeyNav" href="#img${parseInt(imagesRevealed[href], 10)}" title="click to scroll to original">[RES ignored duplicate link]</a>`);
-		$(elem).after(textFrag);
+		return {
+			options: await gfycat.handleLink('', [], info),
+			siteModule: gfycat,
+		};
+	} catch (e) {
+		throw new Error(`Could not convert gif to video ${options.src}: ${e}`);
 	}
 }
 
-function removeRedditExpandoButton(elem) {
-	if (elem.classList.contains('title')) {
-		// remove the default reddit expando button on post listings
-		// we actually remove it from the DOM for a number of reasons, including the
-		// fact that many subreddits style them with display: block !important;, which
-		// overrides a "hide" call here.
-		$(elem).closest('.entry').find('.expando-button.video:not(.commentImg)').remove();
+function getMediaInfo(mediaLink) {
+	let detectResult;
+	const siteModule = [
+		'default',
+		'defaultVideo',
+		'defaultAudio',
+		domainToModuleName(mediaLink.hostname),
+	]
+		.map(modId => domainModuleMap[modId])
+		.find(mod => mod && (detectResult = mod.detect(mediaLink)));
```

Comment:
While we're at it with the generators, we could probably make this a bit cleaner:

```js
const [siteModule, detectResult] = [
    'default',
    'defaultVideo',
    'defaultAudio',
    domainToModuleName(mediaLink.hostname),
]
    ::map(modId => domainModuleMap[modId])
    ::filter(mod => mod)
    ::map(mod => [mod, mod.detect(mediaLink)])
    ::find(([, result]) => result) || [];
```

...actually that might be worse :confused: